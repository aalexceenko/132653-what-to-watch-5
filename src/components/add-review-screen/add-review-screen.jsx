import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import {filmType} from '../../types/film';
import withAddReviewScreen from "../../hocs/with-add-review-screen/with-add-review-screen";
import UserBlock from "../user-block/user-block";
import {postReview} from "../../store/api-action";
import {connect} from "react-redux";
import Logo from "../logo/logo";


const AddReviewScreen = ({textReview, handleSubmit, handleTextChange, handleStarClick, match, films, disabledButton, isError}) => {

  const id = Number(match.params.id);
  const currentFilm = films.find((film) => film.id === id);
  const {title, poster, backgroundImage} = currentFilm;

  return (
    <section className="movie-card movie-card--full">
      <div className="movie-card__header">
        <div className="movie-card__bg">
          <img src={backgroundImage} alt={title} />
        </div>

        <h1 className="visually-hidden">WTW</h1>

        <header className="page-header">
          <Logo />

          <nav className="breadcrumbs">
            <ul className="breadcrumbs__list">
              <li className="breadcrumbs__item">
                <Link to={`/films/${id}`} className="breadcrumbs__link">{title}</Link>
              </li>
              <li className="breadcrumbs__item">
                <a className="breadcrumbs__link">Add review</a>
              </li>
            </ul>
          </nav>

          <UserBlock />
        </header>

        <div className="movie-card__poster movie-card__poster--small">
          <img src={poster} alt={title} width="218" height="327" />
        </div>
      </div>
      <div className="add-review">
        <form action="" className={`add-review__form ${isError ? `shake` : ``} `} onSubmit={handleSubmit}>
          <div className="rating">
            <div className="rating__stars">
              <input className="rating__input" id="star-1" type="radio" name="rating" value="1" onChange={handleStarClick}/>
              <label className="rating__label" htmlFor="star-1">Rating 1</label>

              <input className="rating__input" id="star-2" type="radio" name="rating" value="2" onChange={handleStarClick} />
              <label className="rating__label" htmlFor="star-2">Rating 2</label>

              <input className="rating__input" id="star-3" type="radio" name="rating" value="3" defaultChecked onChange={handleStarClick}/>
              <label className="rating__label" htmlFor="star-3">Rating 3</label>

              <input className="rating__input" id="star-4" type="radio" name="rating" value="4" onChange={handleStarClick}/>
              <label className="rating__label" htmlFor="star-4">Rating 4</label>

              <input className="rating__input" id="star-5" type="radio" name="rating" value="5" onChange={handleStarClick}/>
              <label className="rating__label" htmlFor="star-5">Rating 5</label>
            </div>
          </div>

          <div className="add-review__text">
            <textarea className="add-review__textarea" value={textReview} name="review-text" id="review-text" placeholder="Review text"
              onChange={handleTextChange} />
            <div className="add-review__submit">
              <button className="add-review__btn" disabled={disabledButton} type="submit">Post</button>
            </div>

          </div>
        </form>
      </div>

    </section>
  );

};


AddReviewScreen.propTypes = {
  films: PropTypes.arrayOf(filmType).isRequired,
  match: PropTypes.object.isRequired,
  textReview: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleTextChange: PropTypes.func.isRequired,
  handleStarClick: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  disabledButton: PropTypes.bool.isRequired,
  isError: PropTypes.bool.isRequired,
};

const mapStateToProps = ({APP_PROCESS}) => ({
  films: APP_PROCESS.films,
});

const mapDispatchToProps = (dispatch) => ({
  onSubmit: (ratingStarsChecked, textReview, id) => {
    return dispatch(postReview(ratingStarsChecked, textReview, id));

  },
});

export default connect(mapStateToProps, mapDispatchToProps)(withAddReviewScreen(AddReviewScreen));
