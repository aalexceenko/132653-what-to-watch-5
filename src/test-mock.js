import {getFilmRank} from "./utils";

export const VISIBLE_FILMS_COUNT = 8;
export const ID = 1;
export const VIDEO_CURRENT_TIME = 0;
export const PROGRESS_POSITION = 0;
export const AUTHORIZATION_STATUS = `AUTH`;

export const IS_VIDEO = true;
export const IS_PLAYING = true;
export const IS_IN_MY_LIST = false;
export const TAB = `details`;
export const PREVIEW = `img/aviator.jpg`;
export const PREVIEW_VIDEO = `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;
export const TITLE = `Bohemian Rhapsody`;
export const ACTIVE_GENRE = `Comedy`;
export const GENRES = [`Comedy`, `Crime`, `Documentary`, `Drama`, `Horror`, `Kids & Family`];
export const MATCH = {
  params: {
    id: 1
  }
};

export const FILMS = [
  {
    description: `A young man who was sentenced to seven years`,
    rating: 3.6,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    id: 1,
    title: `Bronson`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    votes: 109661,
    runtime: 92,
    year: 2008,
    ratingText: getFilmRank(3.6),
    myList: false,
    comments: false,
  },
  {
    description: `A young man who was`,
    rating: 3.6,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    id: 2,
    title: `Bronson`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    votes: 109661,
    runtime: 92,
    year: 2008,
    ratingText: getFilmRank(3.6),
    myList: false,
    comments: false,
  },
];

export const FILM = {
  description: `A young man who was sentenced to seven years`,
  rating: 3.6,
  director: `Nicolas Winding Refn`,
  genre: `Action`,
  id: 3,
  title: `Bronson`,
  actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  votes: 109661,
  runtime: 92,
  year: 2008,
  ratingText: getFilmRank(3.6),
  myList: false,
  comments: false,
};

export const REVIEWS = [
  {
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    user: {
      name: `Christina`,
      id: 1,
    },
    comment: `This film is an experience and i has already seen it 4 times`,
  },
  {
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    user: {
      name: `Christina`,
      id: 2,
    },
    comment: `This film is an experience and i has already seen it 4 times`,
  }
];
