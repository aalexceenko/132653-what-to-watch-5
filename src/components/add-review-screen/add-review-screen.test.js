import React from "react";
import renderer from "react-test-renderer";
import {AddReviewScreen} from "./add-review-screen";
import {FILMS, ID, MATCH, TITLE} from "../../test-mock";
import {BrowserRouter} from 'react-router-dom';

jest.mock(`../user-block/user-block`, () => `UserBlock`);
it(`Should AddReviewScreen render correctly`, () => {
  const tree = renderer
    .create(
        <BrowserRouter >
          <AddReviewScreen
            films={FILMS}
            match={MATCH}
            textReview={TITLE}
            handleSubmit={() => {}}
            handleTextChange={() => {}}
            handleStarClick={() => {}}
            id={ID}
            disabledButton={true}
            isError={false}
          />
        </BrowserRouter>

    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
