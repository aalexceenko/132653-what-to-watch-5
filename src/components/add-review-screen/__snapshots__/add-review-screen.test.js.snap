// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should AddReviewScreen render correctly 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Bronson"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link "
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="/films/1"
              onClick={[Function]}
            >
              Bronson
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <UserBlock />
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="Bronson"
        height="327"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      action=""
      className="add-review__form  "
      onSubmit={[Function]}
    >
      <div
        className="rating"
      >
        <div
          className="rating__stars"
        >
          <input
            className="rating__input"
            id="star-1"
            name="rating"
            onChange={[Function]}
            type="radio"
            value="1"
          />
          <label
            className="rating__label"
            htmlFor="star-1"
          >
            Rating 1
          </label>
          <input
            className="rating__input"
            id="star-2"
            name="rating"
            onChange={[Function]}
            type="radio"
            value="2"
          />
          <label
            className="rating__label"
            htmlFor="star-2"
          >
            Rating 2
          </label>
          <input
            className="rating__input"
            defaultChecked={true}
            id="star-3"
            name="rating"
            onChange={[Function]}
            type="radio"
            value="3"
          />
          <label
            className="rating__label"
            htmlFor="star-3"
          >
            Rating 3
          </label>
          <input
            className="rating__input"
            id="star-4"
            name="rating"
            onChange={[Function]}
            type="radio"
            value="4"
          />
          <label
            className="rating__label"
            htmlFor="star-4"
          >
            Rating 4
          </label>
          <input
            className="rating__input"
            id="star-5"
            name="rating"
            onChange={[Function]}
            type="radio"
            value="5"
          />
          <label
            className="rating__label"
            htmlFor="star-5"
          >
            Rating 5
          </label>
        </div>
      </div>
      <div
        className="add-review__text"
      >
        <textarea
          className="add-review__textarea"
          id="review-text"
          name="review-text"
          onChange={[Function]}
          placeholder="Review text"
          value="Bohemian Rhapsody"
        />
        <div
          className="add-review__submit"
        >
          <button
            className="add-review__btn"
            disabled={true}
            type="submit"
          >
            Post
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;
